blockchain/chain.go

```diff
@@ -1385,7 +1380,7 @@ func New(config *Config) (*BlockChain, error) {
 		db:                  config.DB,
 		chainParams:         params,
 		timeSource:          config.TimeSource,
-		notifications:       config.Notifications,
+		notifications:       make([]NotificationCallback, 0),
```

Comment:
For future reference, there is no need to perform an extra unnecessary allocation here.  The `range` and `append` statements in Go work correctly with nil slices.  I'm not going to hold this PR up over it as there are a few other minor nitpicks that I figure I'll just push in a separate PR rather than bugging you about them (`Fatalf` in tests, making test function comment match the function name, test print exceeding col 80, avoiding deferred unlocks in non-trival cases since it has a pretty high overhead)